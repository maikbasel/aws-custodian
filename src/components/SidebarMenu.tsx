'use client';

import Link from "next/link";
import useNavigation from "@/hooks/Navigation";
import {Icon} from "@iconify/react";

import React from 'react';

type MenuItemProps = {
    href: string;
    isActive: boolean;
    iconActive: string;
    iconInactive: string;
    label: string;
};

const MenuItem = ({href, isActive, iconActive, iconInactive, label}: MenuItemProps) => {
    return (
        <Link href={href}
              className="flex flex-row space-x-4 items-center px-4 py-3 duration-200 hover:bg-white/10 w-full" >
            {isActive ? (<Icon icon={iconActive} width="38" height="38" />) : <Icon icon={iconInactive} width="38" height="38" />}
            <span className={`text-2xl pt-2 hidden md:flex ${isActive ? 'font-bold' : ''}`}>{label}</span>
        </Link>
    )
}

export default function SidebarMenu() {
    const {
        isHomeActive,
        isProfilesActive,
    } = useNavigation();

    return (
        <nav
            className="flex-col space-y-4 items-center py-8 hidden sm:flex border-r border-zinc-700 h-full  w-[120px] md:w-[250px] md:items-start fixed bg-gray-700">
            <MenuItem href="/" isActive={isHomeActive} iconActive="ant-design:home-filled"
                            iconInactive="ant-design:home-outlined" label="Home"/>
            <MenuItem href="/profiles" isActive={isProfilesActive} iconActive="ant-design:profile-filled"
                            iconInactive="ant-design:profile-outlined" label="Profiles"/>
        </nav>
    );
}