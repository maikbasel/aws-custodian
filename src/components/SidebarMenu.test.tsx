import React from 'react';
import { render, screen } from '@testing-library/react';
import SidebarMenu from './SidebarMenu';
import Link from 'next/link';
import { Icon } from '@iconify/react';
import renderer from 'react-test-renderer';
import useNavigation from "@/hooks/Navigation";

jest.mock('@/hooks/Navigation');
const mockUseNavigation = useNavigation as jest.MockedFunction<typeof useNavigation>

describe('<SidebarMenu />', () => {


  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render the Home menu item', () => {
    mockUseNavigation.mockReturnValue({
      isHomeActive: false,
      isProfilesActive: false,
    });

    render(<SidebarMenu />);

    const homeMenuItem = screen.getByText('Home');
    expect(homeMenuItem).toBeInTheDocument();
  });

  it('should render the Profiles menu item', () => {
    mockUseNavigation.mockReturnValue({
      isHomeActive: false,
      isProfilesActive: false,
    });

    render(<SidebarMenu />);

    const profilesMenuItem = screen.getByText('Profiles');
    expect(profilesMenuItem).toBeInTheDocument();
  });

  it('should render inactive Home icon when isHomeActive is false', () => {
    mockUseNavigation.mockReturnValue({
      isHomeActive: false,
      isProfilesActive: false,
    });

    const tree = renderer
      .create(
        <Link
          href='/'
          className='flex w-full flex-row items-center space-x-4 px-4 py-3 duration-200 hover:bg-white/10'
        >
          <Icon icon='ant-design:home-outlined' width='38' height='38' />
          <span className='hidden pt-2 text-2xl md:flex'>Home</span>
        </Link>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render active Home icon when isHomeActive is true', () => {
    mockUseNavigation.mockReturnValue({
      isHomeActive: true,
      isProfilesActive: false,
    });

    const tree = renderer
      .create(
        <Link
          href='/'
          className='flex w-full flex-row items-center space-x-4 px-4 py-3 duration-200 hover:bg-white/10'
        >
          <Icon icon='ant-design:home-filled' width='38' height='38' />
          <span className='hidden pt-2 text-2xl md:flex font-bold'>Home</span>
        </Link>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render inactive Profiles icon when isProfilesActive is false', () => {
    mockUseNavigation.mockImplementation(() => ({
      isHomeActive: false,
      isProfilesActive: false,
    }));

    const tree = renderer
      .create(
        <Link
          href='/'
          className='flex w-full flex-row items-center space-x-4 px-4 py-3 duration-200 hover:bg-white/10'
        >
          <Icon icon='ant-design:profile-outlined' width='38' height='38' />
          <span className='hidden pt-2 text-2xl md:flex'>Home</span>
        </Link>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render active Profiles icon when isProfilesActive is true', () => {
    mockUseNavigation.mockReturnValue({
      isHomeActive: false,
      isProfilesActive: true,
    });

    const tree = renderer
      .create(
        <Link
          href='/'
          className='flex w-full flex-row items-center space-x-4 px-4 py-3 duration-200 hover:bg-white/10'
        >
          <Icon icon='ant-design:profile-filled' width='38' height='38' />
          <span className='hidden pt-2 text-2xl md:flex font-bold'>Home</span>
        </Link>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Home menu item with correct href', () => {
    render(<SidebarMenu />);

    const homeMenuItem = screen.getByRole('link', { name: /home/i });
    expect(homeMenuItem).toHaveAttribute('href', '/');
  });

  it('should render Profiles menu item with correct href', () => {
    render(<SidebarMenu />);

    const profilesMenuItem = screen.getByRole('link', { name: /profiles/i });
    expect(profilesMenuItem).toHaveAttribute('href', '/profiles');
  });
});
