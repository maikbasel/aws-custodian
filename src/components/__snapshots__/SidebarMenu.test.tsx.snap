// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarMenu /> should render active Home icon when isHomeActive is true 1`] = `
<a
  className="flex w-full flex-row items-center space-x-4 px-4 py-3 duration-200 hover:bg-white/10"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <span />
  <span
    className="hidden pt-2 text-2xl md:flex font-bold"
  >
    Home
  </span>
</a>
`;

exports[`<SidebarMenu /> should render active Profiles icon when isProfilesActive is true 1`] = `
<a
  className="flex w-full flex-row items-center space-x-4 px-4 py-3 duration-200 hover:bg-white/10"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <span />
  <span
    className="hidden pt-2 text-2xl md:flex font-bold"
  >
    Home
  </span>
</a>
`;

exports[`<SidebarMenu /> should render inactive Home icon when isHomeActive is false 1`] = `
<a
  className="flex w-full flex-row items-center space-x-4 px-4 py-3 duration-200 hover:bg-white/10"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <span />
  <span
    className="hidden pt-2 text-2xl md:flex"
  >
    Home
  </span>
</a>
`;

exports[`<SidebarMenu /> should render inactive Profiles icon when isProfilesActive is false 1`] = `
<a
  className="flex w-full flex-row items-center space-x-4 px-4 py-3 duration-200 hover:bg-white/10"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <span />
  <span
    className="hidden pt-2 text-2xl md:flex"
  >
    Home
  </span>
</a>
`;
